library/src/main/java/com/google/android/exoplayer/hls/HlsChunkSource.java

```diff
@@ -573,7 +624,16 @@ private void setMediaPlaylist(int variantIndex, HlsMediaPlaylist mediaPlaylist)
     variantLastPlaylistLoadTimesMs[variantIndex] = SystemClock.elapsedRealtime();
     variantPlaylists[variantIndex] = mediaPlaylist;
     live |= mediaPlaylist.live;
-    durationUs = live ? C.UNKNOWN_TIME_US : mediaPlaylist.durationUs;
+    liveEvent |= mediaPlaylist.liveEvent;
+    durationUs = live || liveEvent ? C.UNKNOWN_TIME_US : mediaPlaylist.durationUs;
+
+    if(!live || liveEvent) {
```

Comment:
Why is "live" being filtered out here.  I would think that both a sliding fixed size live window and a growing sized live window (is this the same as an "Event" type?) would want to notify listeners about the available seek range change.